src/main/java/macrobase/analysis/pipeline/Pipeline.java

```diff
@@ -0,0 +1,9 @@
+package macrobase.analysis.pipeline;
+
+import macrobase.analysis.result.AnalysisResult;
+import macrobase.conf.MacroBaseConf;
+
+public interface Pipeline {
```

Comment:
A once sentence description what each method is for. It took me some time to understand the new iterator-like interfaces so having something as a documentation would be helpful.